### PR TITLE
Throw EDQUOT error when ufs quota exceeded

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?
When using ozone or hdfs as ufs, fuse throws EDQUOT instead of EIO when quota exceeded.

### Why are the changes needed?
Quota exceeded exception can not be thrown.

### Does this PR introduce any user facing changes?
NA.

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java [hunk 1]
```diff
@@ -96,6 +97,9 @@ public final class AlluxioFuseUtils {
   public static final long MODE_NOT_SET_VALUE = -1;
   public static final String MACFUSE_SUPPORT_MINIMUM_OS_VERSION = "10.9";
 
+  private static final Pattern QUOTA_EXCEEDED_PATTERN =
+      Pattern.compile("(.*)The DiskSpace quota of(.*)exceeded([\\s\\S]*)");
```
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:101]: Is it true others would return the same error message? Or there is a way to check according to a return value other than a string match?

Comment (Haoning-Sun) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:101]: The pattern is only applicable to ozone and hdfs. Unable to identify the exception classes of ozone and hdfs, so check the message.

Comment (yyongycy) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:101]: could it be converted to a standard type in the UFS? The message matching is not a good way.

**Comment by LuQQiu** [comment]
The Fuse side exception will need to be redone to translate exceptions to FUSE return errors

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java (hunk 1 above)
Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:101]: @Haoning-Sun can you help give us the whole workflow of where this exception is thrown originally and how it's reach to FUSE in the end, and why this exception is thrown (when what happens). This help us figure out whether they are better ways to pass the original exception (may be a better exception that refers to quota exceed) to FUSE and directly translate this exception to error code

Comment (Haoning-Sun) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:101]: - Create bucket and volume in ozone and use 'setquota' to set space quota for volume.
- Mount the ozone location in the alluxio.
- Start a fuse process which its write type is CACHE_THROUGH  to mount the alluxio path to a local directory.
- Copy a file exceeding the quota to the mount directory. The following exception will be thrown and return EIO.
```
2023-01-05 22:45:53,236 ERROR AlluxioFuseUtils - Failed to Fuse.Write(path=/test.txt,fd=1,size=65536,offset=11468800) with unexpected throwable:
java.lang.RuntimeException: alluxio.exception.status.UnknownException: The DiskSpace quota of bucket:testexceeded: quotaInBytes: 104857600 Bytes but diskspace consumed: 805306368 Bytes. (GrpcDataWriter{request=type: UFS_FILE
...
```

**Comment by Haoning-Sun** [comment]
@LuQQiu This is a good way. Maybe we can discuss how to handle these UnknownException.

**Comment by LuQQiu** [comment]
@Haoning-Sun Sounds good! Feel free to find me at Slack or Wechat.
Ideally we want Alluxio to throw more clear exception at the original place as a subclass of AlluxioRuntimeException. And the exception will be all the way thrown to FUSE side and be caught by FUSE. FUSE based on the subclass exception type, translates to the corresponding error code.
@jja725 refactor many of the exception logics, but we are kind of worried about breaking the backward compatibility so do not push it to be executed throughout the repo.
But in the new alluxio dora branch, especially for FUSE, we try to follow this approach

**Pull request closed without merging**